vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,321 @@
+
+-   **Rhino will not change your working directory** - You need to either open a new R session in your new application directory or manually change the working directory.
+
+```         
+setwd("./RhinoApplication")
+```
+
+-   **Rhino relies on options added to the projects `.Rprofile` file -** The most robust way to make sure it was correctly sourced is to simply restart the R session.
+
+A result of both paths will be an initial Rhino application with the following structure:
+
+```         
+.
+├── RhinoApplication.Rproj
+├── app
+│ ├── js
+│ │ └── index.js
+│ ├── logic
+│ │ └── __init__.R
+│ ├── main.R
+│ ├── static
+│ │ └── favicon.ico
+│ ├── styles
+│ │ └── main.scss
+│ └── view
+│     └── __init__.R
+├── app.R
+├── config.yml
+├── dependencies.R
+├── renv.lock
+├── rhino.yml
+└── tests
+    ├── cypress
+    │ └── integration
+    │     └── app.spec.js
+    ├── cypress.json
+    └── testthat
+        └── test-main.R
+```
+
+If you want to know more about it, check [this](https://appsilon.github.io/rhino/articles/explanation/application-structure.html) document.
+
+### 🚀 Running the initial application
+
+Now, once you are all set up, let's run it:
+
+```         
+shiny::runApp()
+```
+
+And here is what you should be seeing right now 👇
+
+![](https://appsilon.github.io/rhino/articles/tutorial/images/init_application.png){width="620" height="268"}
+
+## Let's install `shiny.fluent` for our project
+
+💡The rhino project you just started uses `renv` for dependency management.
+You can now install any other package in the same project and include it in the `renv.lock` file.
+
+Here's how you can install the `shiny.fluent` package 👇
+
+``` r
+install.packages("shiny.fluent")
+```
+
+Now that the package is installed, let us use it for building our very first `rhino` module.
+
+## Use `shiny.fluent` in your Rhino application!
+
+In Rhino, each application view is intended to live as a [Shiny module](https://shiny.rstudio.com/articles/modules.html) and use encapsulation provided by [`box` package](https://klmr.me/box/).
+Also, in rhino apps, even the base ui and server are modules!
+
+In your project, please go to the `app` folder and edit the `main.R` file with the code below
+
+``` r
+# app/main.R
+
+box::use(
+  shiny[bootstrapPage, moduleServer, NS, tags, textOutput, renderText],
+  shiny.fluent[fluentPage]
+)
+
+#' @export
+# Define user interface function
+ui <- function(id) {
+  # Create namespace for UI elements
+  ns <- NS(id)
+  # Define app layout using Fluent UI design system
+  fluentPage(
+    tags$h1(
+      textOutput(ns("message")) # Display message generated by server
+    )
+  )
+}
+
+#' @export
+# Define server logic function
+server <- function(id) {
+  # Create Shiny module
+  moduleServer(id, function(input, output, session) {
+    # Generate message to be displayed in UI
+    output$message <- renderText("👋 This text has been generated using shiny.fluent package inside a Rhino project!")
```

Comment:
This message is not generated using shiny.fluent. This is just a basic Shiny functionality.